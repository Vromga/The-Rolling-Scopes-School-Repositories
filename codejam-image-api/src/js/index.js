import '../css/style.css';
import '../css/style.scss';
import { toggleActiveClass, tools } from "./module/toggleActiveClass";
import showMessage from "./module/showMessage";
import grayscale from "./module/grayscale";
import clearCanvas from "./module/clearCanvas";
import takeColor from "./module/takeColor";
import colorSave from "./module/colorSave";
import rgbToHex from "./module/rgbToHex";

const canvas = document.querySelector('#canvas');
const ctx = canvas.getContext('2d');
const searchInput = document.querySelector('#search');
const resolutionSection = document.querySelectorAll('.resolution--button');
const clientID = '6eead276a92ca5c7033f38dccfea6eb3aa045a61c5789af291ecd46ceda74be5';

function removeResolutionClass() {
    [...resolutionSection].forEach((elem) => {
        elem.classList.remove('active');
    });
}

let isDraw = false;
let isInput = false;
let isGrayscale = true;
searchInput.blur();

function resetToolsState() {
    for (const prop of Object.keys(tools)) {
        tools[prop] = false;
    }
}

toggleActiveClass();
const realSizeCanvas = 512;
let virtualPixelCanvas = 4;
canvas.style.width = `${realSizeCanvas}px`;
canvas.style.height = `${realSizeCanvas}px`;
canvas.width = realSizeCanvas / virtualPixelCanvas;
canvas.height = realSizeCanvas / virtualPixelCanvas;
ctx.fillStyle = 'rgba(255,255,255,255)';
ctx.fill();
ctx.fillRect(0, 0, canvas.width, canvas.height);

function saveApp() {
    localStorage.setItem('canvasImage', canvas.toDataURL());
    localStorage.setItem('tools', JSON.stringify(tools));
    localStorage.setItem('color', colorSave());
    localStorage.setItem('pixel', JSON.stringify(virtualPixelCanvas));
}

function restoreCanvas() {
    const pixel = +JSON.parse(localStorage.getItem('pixel'));
    removeResolutionClass();
    if (pixel === 1) {
        [...resolutionSection][3].classList.add('active');
    } else if (pixel === 2) {
        [...resolutionSection][2].classList.add('active');
    } else if (pixel === 4) {
        [...resolutionSection][1].classList.add('active');
    } else if (pixel === 8) {
        [...resolutionSection][0].classList.add('active');
    }
    changeSize(pixel);
    const dataURL = localStorage.getItem('canvasImage');
    const img = new Image();
    img.src = dataURL;
    img.onload = () => {
        ctx.drawImage(img, 0, 0);
    };

    const toolsJSON = JSON.parse(localStorage.getItem('tools'));
    resetToolsState();
    tools.fillBucket = toolsJSON.fillBucket;
    tools.chooseColor = toolsJSON.chooseColor;
    tools.pencil = toolsJSON.pencil;
    toggleActiveClass();

    const colorJSON = JSON.parse(localStorage.getItem('color'));
    document.querySelector('#current').value = colorJSON.current;
    document.querySelector('#prev').style.backgroundColor = colorJSON.prev;
}

function searchWord() {
    return searchInput.value.split(' ').join(',');
}

function setPrevColor() {
    const temp = takeColor();
    const prevColor = document.querySelectorAll('.prev');
    const circlePrevColor = document.querySelector('#prev');
    circlePrevColor.style.backgroundColor = temp;
    [...prevColor].map((element) => element.setAttribute('dataColor', `${temp}`));
}

const getColorAtPixel = (imageData, x, y) => {
    const { width, data } = imageData;

    return {
        r: data[4 * (width * y + x)],
        g: data[4 * (width * y + x) + 1],
        b: data[4 * (width * y + x) + 2],
        a: data[4 * (width * y + x) + 3],
    };
};

async function getLinkToImage(search) {
    const url = `https://api.unsplash.com/photos/random?query=${search}&client_id=${clientID}`;
    const response = await fetch(url);
    const data = await response.json();
    const img = new Image();
    img.src = data.urls.regular;
    img.crossOrigin = 'Anonymous';
    return img;
}

function loadImage(img) {
    img.onload = function f() {
        let ratio;
        let width;
        let height;
        let x;
        let y;

        function checkSizeImage() {
            if (img.width > img.height) {
                ratio = img.width / img.height;
                width = canvas.width;
                height = width / ratio;
            } else {
                ratio = img.height / img.width;
                height = canvas.height;
                width = height / ratio;
            }
        }

        function alignCenter() {
            x = canvas.width / 2 - width / 2;
            y = canvas.height / 2 - height / 2;
        }

        function drawImg(img, x, y, w, h) {
            ctx.drawImage(img, x, y, w, h);
        }

        checkSizeImage();
        alignCenter();
        drawImg(img, x, y, width, height);
    };
}

function selectTools(event) {
    resetToolsState();
    if (event.target.classList.contains('fillBucket') || event.code === 'KeyB') {
        tools.fillBucket = true;
        showMessage('selected tool fill bucked');
    } else if (event.target.classList.contains('chooseColor') || event.code === 'KeyC') {
        tools.chooseColor = true;
        showMessage('selected tool pencil');
    } else if (event.target.classList.contains('pencil') || event.code === 'KeyP') {
        tools.pencil = true;
        showMessage('selected tool choose color');
    }
    toggleActiveClass();
}

function selectColor(event) {
    const current = document.querySelector('#current');
    if (event.target.getAttribute('data') === 'blue') {
        setPrevColor();
        current.value = '#0000FF';
    } else if (event.target.getAttribute('data') === 'red') {
        setPrevColor();
        current.value = '#FF0000';
    } else if (event.target.getAttribute('data') === 'prev') {
        current.value = event.target.previousElementSibling.getAttribute('dataColor');
    } else if (event.target.getAttribute('data') === 'current' || event.target.type === 'color') {
        setPrevColor();
    }
}

function selectResolution(e) {
    if (e.target.getAttribute('data') === '512') {
        removeResolutionClass();
        e.target.classList.add('active');
        changeSize(1);
        showMessage('selected resolution canvas 512px');
    }

    if (e.target.getAttribute('data') === '256') {
        removeResolutionClass();
        e.target.classList.add('active');
        changeSize(2);
        showMessage('selected resolution canvas 256px');
    }

    if (e.target.getAttribute('data') === '128') {
        removeResolutionClass();
        e.target.classList.add('active');
        changeSize(4);
        showMessage('selected resolution canvas 128px');
    }

    if (e.target.getAttribute('data') === '64') {
        removeResolutionClass();
        e.target.classList.add('active');
        changeSize(8);
        showMessage('selected resolution canvas 64px');
    }

    if (e.target.className === 'tools--buttons-elem') {
        selectTools(e);
    }
}

const cordPixel = {
    x0: 0,
    y0: 0,
    x1: 0,
    y1: 0,
};

function setPixel(x, y) {
    ctx.beginPath();
    ctx.fillStyle = `${takeColor()}`;
    ctx.fill();
    ctx.fillRect(x, y, 1, 1);
    cordPixel.x0 = cordPixel.x1;
    cordPixel.y0 = cordPixel.y1;
}

function algoritmBresenham(x0, y0, x1, y1) {
    const dx = Math.abs(x1 - x0);
    const sx = x0 < x1 ? 1 : -1;
    const dy = Math.abs(y1 - y0);
    const sy = y0 < y1 ? 1 : -1;
    let err = (dx > dy ? dx : -dy) / 2;
    while (true) {
        setPixel(x0, y0);
        if (x0 === x1 && y0 === y1) break;
        const e2 = err;
        if (e2 > -dx) {
            err -= dy;
            x0 += sx;
        }
        if (e2 < dy) {
            err += dx;
            y0 += sy;
        }
    }
}

function changeSize(n) {
    localStorage.setItem('canvas', canvas.toDataURL());
    clearCanvas();
    virtualPixelCanvas = n;
    canvas.width = realSizeCanvas / virtualPixelCanvas;
    canvas.height = realSizeCanvas / virtualPixelCanvas;
    const dataURL = localStorage.getItem('canvas');
    const img = new Image();
    img.src = dataURL;
    loadImage(img);
}

document.addEventListener('mousedown', (event) => {
            if (tools.pencil === true && event.target.id === 'canvas') {
                isDraw = true;
                cordPixel.x0 = Math.floor(event.offsetX / (realSizeCanvas / canvas.width));
                cordPixel.y0 = Math.floor(event.offsetY / (realSizeCanvas / canvas.height));
            } else if (tools.chooseColor === true && event.target.id === 'canvas') {
                const x = Math.floor(event.offsetX / (realSizeCanvas / canvas.width));
                const y = Math.floor(event.offsetY / (realSizeCanvas / canvas.height));
                const current = document.querySelector('#current');
                const colorPick = ctx.getImageData(0, 0, realSizeCanvas, realSizeCanvas);
                setPrevColor();
                const objColor = getColorAtPixel(colorPick, x, y);
                current.value = `#${(`000000${rgbToHex(objColor.r, objColor.g, objColor.b)}`).slice(-6)}`;
  } else if (tools.fillBucket === true && event.target.id === 'canvas') {
    ctx.fillStyle = `${takeColor()}`;
    ctx.fill();
    ctx.fillRect(0, 0, canvas.width, canvas.height);
  }
});
document.addEventListener('mousemove', (event) => {
  if (isDraw === true) {
    cordPixel.x1 = Math.floor(event.offsetX / (realSizeCanvas / canvas.width));
    cordPixel.y1 = Math.floor(event.offsetY / (realSizeCanvas / canvas.height));
    algoritmBresenham(cordPixel.x0, cordPixel.y0, cordPixel.x1, cordPixel.y1);
  }
});
document.addEventListener('mouseup', () => {
  if (tools.pencil === true) {
    isDraw = false;
  }
});
document.addEventListener('click', (e) => {
  if (tools.pencil === true && e.target.id === 'canvas') {
    const x = Math.floor(e.offsetX / (realSizeCanvas / canvas.width));
    const y = Math.floor(e.offsetY / (realSizeCanvas / canvas.height));
    ctx.fillStyle = `${takeColor()}`;
    ctx.fill();
    ctx.fillRect(x, y, 1, 1);
  }
  isInput = e.target.id === 'search';

  if (e.target.innerText === 'load') {
    clearCanvas();
    const searchQuery = searchWord();
    getLinkToImage(searchQuery).then((value) => loadImage(value));
  }

  selectColor(e);

  if (e.target.getAttribute('data') === 'button') selectTools(e);

  selectResolution(e);

  if (e.target.innerText === 'b/w') {
    if (isGrayscale) {
      localStorage.setItem('canvas', canvas.toDataURL());
      grayscale();
      isGrayscale = false;
    } else {
      const dataURL = localStorage.getItem('canvas');
      const img = new Image();
      img.src = dataURL;
      img.onload = function imgDraw() {
        ctx.drawImage(img, 0, 0);
      };
      isGrayscale = true;
    }
  }
});
document.addEventListener('keydown', (event) => {
  if (isInput === false) {
    selectTools(event);
    switch (event.code) {
      case 'KeyS':
        saveApp();
        showMessage('app save');
        break;

      case 'KeyR':
        clearCanvas();
        showMessage('canvas clear');
        break;

      case 'KeyL':
        restoreCanvas();
        showMessage('app load');
        break;

      default:
        break;
    }
  }
});
document.addEventListener('keyup', (event) => {
  if (event.code === 'Enter') {
    const loadButton = document.querySelector('.canvas--load');
    loadButton.click();
  }
});