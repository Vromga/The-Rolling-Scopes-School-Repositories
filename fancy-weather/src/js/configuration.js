const lang = [
  'EN', 'RU', 'BY'
];

const keyWeather = '36991b1be3297e1561963c6ed47c155e';
const keyGeoLocation = '1db8194a2703a0';
const imageKey = 'f258bef5c6ebd9401bba43395eb2fd21';
const keyOpenData = '65467207ccbd4bf1a8d0f1e6c215bbbf';

const optionMap = {
  container: 'map',
  center: [-79.4512, 43.6568],
  zoom: 13,
  style: 'mapbox://styles/vromga/ck3x27sr35ddl1cmsxvf1zuf6',
};



export {lang, keyWeather, keyGeoLocation, imageKey, optionMap, keyOpenData};
