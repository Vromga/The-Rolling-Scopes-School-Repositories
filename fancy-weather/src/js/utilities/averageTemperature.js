function averageTemperature(max, min) {
  return (max + min) / 2;
}

export default averageTemperature;
