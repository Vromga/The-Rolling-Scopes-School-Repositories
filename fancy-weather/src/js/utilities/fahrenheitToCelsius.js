function fahrenheitToCelsius(temp) {
  return Math.trunc((temp - 32) / 1.8);
}

export default fahrenheitToCelsius;
