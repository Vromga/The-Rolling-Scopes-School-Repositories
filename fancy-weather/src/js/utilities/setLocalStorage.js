function setLocalStorage(name, value) {
  localStorage.setItem(name, value);
}
export default setLocalStorage;
