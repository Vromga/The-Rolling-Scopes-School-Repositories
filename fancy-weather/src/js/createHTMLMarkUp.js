import {lang} from './configuration';

function createHTMLMarkUp() {

  const weatherArr = [
    'Feels Like 7C',
    'wind 2ms',
    'Humidity 83%'
  ];


  function createSingleHTMLElement(name, className) {
    const elem = document.createElement(name);
    elem.classList.add(className);
    return elem;
  }

  function createNestedHTMLElement(arr, parent, children, parentClass, childrenClass, attributeName) {
    const selectLang = createSingleHTMLElement(parent, parentClass);
    arr.forEach(value => {
      const optionElem = createSingleHTMLElement(children, childrenClass);
      optionElem.setAttribute(attributeName, `${value}`);
      optionElem.textContent = `${value}`;
      selectLang.append(optionElem);
    });
    return selectLang;
  }


  const body = document.querySelector('body');
  const app = createSingleHTMLElement('div', 'app');

  //      ########## HEADER ############      //

  const header = createSingleHTMLElement('header', 'header');
  const reloadImageButton = createSingleHTMLElement('button', 'header--reload_image');
  const selectLanguage = createNestedHTMLElement(lang, 'select', 'option', 'header--select', 'header--select-option', 'value');
  const buttonSwitchTemperature = createSingleHTMLElement('button', 'header--switch');
  buttonSwitchTemperature.textContent = `F${String.fromCharCode(176)}`;
  const searchWrapper = createSingleHTMLElement('div', 'header--search');
  const inputSearch = createSingleHTMLElement('input', 'header--search-input');
  inputSearch.setAttribute('type', 'search');
  inputSearch.setAttribute('placeholder', 'Search city or ZIP');
  const buttonSearch = createSingleHTMLElement('button', 'header--search-button');
  buttonSearch.textContent = 'search';

  searchWrapper.append(inputSearch);
  searchWrapper.append(buttonSearch);

  header.append(reloadImageButton);
  header.append(selectLanguage);
  header.append(buttonSwitchTemperature);
  header.append(searchWrapper);

  //      ########## MAIN ############      //

  const main = createSingleHTMLElement('main', 'main');

  //      ########## Weather today block ############      //

  const sectionWeatherToday = createSingleHTMLElement('section', 'weather_today');
  const town = createSingleHTMLElement('p', 'weather_today--town');
  const dateToday = createSingleHTMLElement('p', 'weather_today--date');
  const timeNow = createSingleHTMLElement('p', 'weather_today--time');
  const temperatureNow = createSingleHTMLElement('p', 'weather_today--temperature');
  const iconWeatherNow = createSingleHTMLElement('div', 'weather_today--icon');
  const recommendationWeather = createSingleHTMLElement('p', 'weather_today--recommendation');
  const weatherOptionToday = createNestedHTMLElement(weatherArr, 'ul', 'li', 'weather_today--option', 'weather_today--option-elem', 'data');

  sectionWeatherToday.append(town);
  sectionWeatherToday.append(dateToday);
  sectionWeatherToday.append(timeNow);
  sectionWeatherToday.append(temperatureNow);
  sectionWeatherToday.append(iconWeatherNow);
  sectionWeatherToday.append(recommendationWeather);
  sectionWeatherToday.append(weatherOptionToday);
  main.append(sectionWeatherToday);

  //      ########## Weather future block ############      //

  const sectionWeatherFuture = createSingleHTMLElement('section', 'weather_future');
  const tomorrowWeather = createSingleHTMLElement('ul', 'weather_future--elem');
  tomorrowWeather.setAttribute('data', 'day1');
  const tomorrowWeatherWeek = createSingleHTMLElement('li', 'weather_future--elem-week');
  const tomorrowWeatherTemperature = createSingleHTMLElement('li', 'weather_future--elem-temperature');
  const tomorrowWeatherIcon = createSingleHTMLElement('li', 'weather_future--elem-icon');

//      ########## Delete after ############      //
  tomorrowWeatherWeek.textContent = 'Пн';
  tomorrowWeatherTemperature.textContent = `-18${String.fromCharCode(176)}`;
  tomorrowWeatherIcon.style.backgroundImage = 'url("./img/rain.png")';
//      ########## Delete after ############      //

  tomorrowWeather.append(tomorrowWeatherWeek);
  tomorrowWeather.append(tomorrowWeatherTemperature);
  tomorrowWeather.append(tomorrowWeatherIcon);

  const afterTomorrowWeather = tomorrowWeather.cloneNode(true);
  afterTomorrowWeather.setAttribute('data', 'day2');
  const afterTheDayAfterTomorrow = tomorrowWeather.cloneNode(true);
  afterTheDayAfterTomorrow.setAttribute('data', 'day3');
  sectionWeatherFuture.append(tomorrowWeather);
  sectionWeatherFuture.append(afterTomorrowWeather);
  sectionWeatherFuture.append(afterTheDayAfterTomorrow);



  main.append(sectionWeatherFuture);

  //      ########## Map block ############      //

  const sectionMap = createSingleHTMLElement('section', 'map');
  const mapBox = createSingleHTMLElement('div', 'map--map');
  const textCordinates = createSingleHTMLElement('p', 'map--cord-text');
  const latitude = createSingleHTMLElement('p', 'map--cord-lat');
  const longitude = createSingleHTMLElement('p', 'map--cord-lon');


  sectionMap.append(mapBox);
  sectionMap.append(textCordinates);
  sectionMap.append(latitude);
  sectionMap.append(longitude);


  main.append(sectionMap);

  app.append(header);
  app.append(main);

  //      ########## Delete after ############      //

  town.textContent = 'Minsk, Belarus';
  dateToday.textContent = 'Mon 30 Nov 2019';
  timeNow.textContent = '22:58';
  temperatureNow.textContent = `-5${String.fromCharCode(176)}`;
  recommendationWeather.textContent = 'Облачно, холодно, одевайтесь потеплее';
  textCordinates.textContent = 'Your coordinates:';
  latitude.textContent = `Latitude: 53${String.fromCharCode(176)} 28'`;
  longitude.textContent = `Latitude: 28${String.fromCharCode(176)} 08'`;

  body.insertAdjacentElement('afterbegin', app);
}

export default createHTMLMarkUp;
