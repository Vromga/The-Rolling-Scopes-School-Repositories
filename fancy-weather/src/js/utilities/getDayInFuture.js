import {week} from "../configuration";

function getDayInFuture(day) {
  const tomorrowDay = document.querySelector('.weather_future--tomorrow-day');
  const afterTomorrowDay = document.querySelector('.weather_future--after_tomorrow-day');
  const afterAfterTomorrowDay = document.querySelector('.weather_future--after_after_tomorrow-day');

  if (day === 5){
    tomorrowDay.textContent = week.en[6];
    afterTomorrowDay.textContent = week.en[7];
    afterAfterTomorrowDay.textContent = week.en[0];
  }

  if (day === 6){
    tomorrowDay.textContent = week.en[7];
    afterTomorrowDay.textContent = week.en[0];
    afterAfterTomorrowDay.textContent = week.en[1];
  }

  if (day === 7){
    tomorrowDay.textContent = week.en[0];
    afterTomorrowDay.textContent = week.en[1];
    afterAfterTomorrowDay.textContent = week.en[2];
  }

  if (day < 5){
    tomorrowDay.textContent = week.en[day];
    afterTomorrowDay.textContent = week.en[day+1];
    afterAfterTomorrowDay.textContent = week.en[day+2];
  }
}

export default getDayInFuture;
