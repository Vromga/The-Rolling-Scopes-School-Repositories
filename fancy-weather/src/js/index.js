import '../css/style.css';
import '../css/style.scss';
import createHTMLMarkUp from "./DOM/createHTMLMarkUp";
import getImage from "./api/flickrAPI";
import convertTemperatureToCelsius from "./utilities/convertTemperatureToCelsius";
import getGeoLocation from "./api/geoCodingApiReverse";
import getLanguage from "./utilities/getLanguage";
import getLocalStorage from "./utilities/getLocalStorage";
import setChoiceLang from "./utilities/setChoiceLang";
import geoCoding from "./api/geoCodingApiForward";

getLocalStorage();
createHTMLMarkUp();
setChoiceLang();
getGeoLocation();
//getImage();


document.addEventListener('click', (e) => {
  if (e.target.className === 'header--switch') {
    convertTemperatureToCelsius();
  }
});

document.addEventListener('change',(e)=>{
  if(e.target.id === 'searchCity'){
    geoCoding();
  }
  if (e.target.id === 'select-lang'){
    getLanguage();
  }
});
