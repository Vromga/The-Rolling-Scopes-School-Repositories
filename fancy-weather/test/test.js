import celsiusToFahrenheit from "../src/js/utilities/celsiusToFarenheit";
import fahrenheitToCelsius from "../src/js/utilities/fahrenheitToCelsius";
import convertPressure from "../src/js/utilities/converPressure";
import setLocalStorage from "../src/js/utilities/setLocalStorage";
import {mouthBE, UI, keyWeather} from "../src/js/configuration";
import averageTemperature from "../src/js/utilities/averageTemperature";
import getDayInFuture from "../src/js/utilities/getDayInFuture";

it ('temperature celsius to fahrenheit', () =>{
	expect(celsiusToFahrenheit(35)).toEqual(95);
	expect(celsiusToFahrenheit(1)).toEqual(33);
	expect(celsiusToFahrenheit(-11)).toEqual(12);
	expect(celsiusToFahrenheit(17)).toEqual(62);
});

it ('temperature fahrenheit to celsius', () =>{
	expect(fahrenheitToCelsius(60)).toEqual(15);
	expect(fahrenheitToCelsius(34)).toEqual(1);
	expect(fahrenheitToCelsius(0)).toEqual(-17);
	expect(fahrenheitToCelsius(95)).toEqual(35);
});

it ('convert hectopascal to millimeter mercury', () =>{
	expect(convertPressure(1200)).toEqual('900');
	expect(convertPressure(988)).toEqual('741');
	expect(convertPressure(1024)).toEqual('768');
	expect(convertPressure(-1090)).toEqual('-818');
});

it ('setLocalStorage to be undefined', () =>{
	expect(setLocalStorage('city', 'Minsk')).toBeUndefined();
});

it ('mounts be lang is Array', () =>{
	expect(mouthBE).toBeInstanceOf(Array);
});

it ('UI is Object', () =>{
	expect(UI).toBeInstanceOf(Object);
});

it ('keyWeather is String', () =>{
	expect(keyWeather).toEqual(expect.any(String));
});

it ('mounts in be lang array length', () =>{
	expect(mouthBE.length).toEqual(12);
});

it ('mounts in be lang have correct name', () =>{
	expect(mouthBE.includes('Сакавiка')).toBeTruthy();
	expect(mouthBE.includes('Марта')).toBeFalsy();
});

it ('mounts in be lang have correct name', () =>{
	expect(mouthBE.includes('Сакавiка')).toBeTruthy();
	expect(mouthBE.includes('Марта')).toBeFalsy();
});

it ('mounts in be lang array have this name', () =>{
	expect(mouthBE).toContain('Лiстапада');
	expect(mouthBE[0]).toEqual('Студзеня');
	expect(mouthBE[5]).toMatch(/Чэрвеня/);
});

it ('setSnapshot', () =>{
	const resultPressure = convertPressure(1200);
	expect(resultPressure).toMatchSnapshot();
	const resultAvgTemper = averageTemperature(42, 30);
	expect(resultAvgTemper).toMatchSnapshot();
});

it ('average temperature', () =>{
	expect(averageTemperature(42, 15)).toEqual(28.5);
	expect(averageTemperature(-5, 2)).toEqual(-1.5);
	expect(averageTemperature(42, 30)).toEqual(36);
	expect(averageTemperature(60, 7)).toEqual(33.5);
});

it ('get day in future', () =>{
	expect(getDayInFuture()).toBeUndefined()
});












