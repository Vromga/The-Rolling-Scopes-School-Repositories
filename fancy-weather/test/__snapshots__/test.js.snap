// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setSnapshot 1`] = `"900"`;

exports[`setSnapshot 2`] = `36`;
